server/controllers/changeNonResidentialType/details.ts

```diff
@@ -0,0 +1,71 @@
+import FormWizard from 'hmpo-form-wizard'
+import { NextFunction, Response } from 'express'
+import FormInitialStep from '../base/formInitialStep'
+
+export default class ChangeNonResidentialTypeDetails extends FormInitialStep {
+  middlewareSetup() {
+    this.use(this.setOptions)
+    super.middlewareSetup()
+  }
+
+  async setOptions(req: FormWizard.Request, res: Response, next: NextFunction) {
+    const token = await req.services.authService.getSystemClientToken(res.locals.user.username)
+    const convertedCellType = await req.services.locationsService.getConvertedCellTypes(token)
+
+    req.form.options.fields.convertedCellType.items = Object.values(convertedCellType).map(({ key, description }) => ({
+      value: key,
+      text: description,
+      conditional: key === 'OTHER' ? 'otherConvertedCellType' : undefined,
+    }))
+
+    next()
+  }
+
+  locals(req: FormWizard.Request, res: Response): Record<string, any> {
+    const locals = super.locals(req, res)
+    const { location } = res.locals
+    const { id: locationId, prisonId } = location
+    const cancelLink = `/view-and-update-locations/${prisonId}/${locationId}`
+
+    return {
+      ...locals,
+      backLink: cancelLink,
+      cancelLink,
+    }
+  }
+
+  async saveValues(req: FormWizard.Request, res: Response, next: NextFunction) {
+    try {
+      const { user } = res.locals
+      const { locationsService } = req.services
+      const token = await req.services.authService.getSystemClientToken(user.username)
+      const { options, values } = req.form
+
+      await locationsService.changeNonResType(
+        token,
+        res.locals.location.id,
+        String(values.convertedCellType),
+        values.convertedCellType === 'OTHER' ? String(values.otherConvertedCellType) : undefined,
```

Comment:
I believe these should always be strings, so `String(...)` is not needed

Reply:
it should be but the backend might accepts an array 
![image](https://github.com/user-attachments/assets/a4ba43ea-abde-48d5-8036-a13f862c6548)